src/test/java/net/snowflake/client/core/SessionUtilWiremockIT.java

```
@@ -231,6 +233,33 @@ public void testOktaRetriesUntilTimeoutThenRaisesAuthTimeoutExceptionWhen429InFe
         ALLOWED_DIFFERENCE_BETWEEN_LOGIN_TIMEOUT_AND_ACTUAL_DURATION_IN_MS);
   }
 
+  @Test
+  public void testErrorHandlingWhenOktaReturnsUnsupportedMfaInFederatedStep3() throws Throwable {
+    // GIVEN
+    Map<String, Object> placeholders = new HashMap<>();
+    placeholders.put("{{WIREMOCK_HOST_WITH_HTTPS_AND_PORT}}", WIREMOCK_HOST_WITH_HTTPS_AND_PORT);
+    String wireMockMapping =
+        getWireMockMappingFromFile(UNSUPPORTED_MFA_IN_FEDERATED_STEP_3, placeholders);
+    importMapping(wireMockMapping);
+
+    setCustomTrustStorePropertyPath();
+
+    SFLoginInput loginInput = createOktaLoginInputBase();
+    Map<SFSessionProperty, Object> connectionPropertiesMap = initConnectionPropertiesMap();
+
+    try {
+      // WHEN
+      SessionUtil.openSession(loginInput, connectionPropertiesMap, "ALL");
+      // THEN
+    } catch (SnowflakeSQLException ex) {
```

It would be better to use `assertThrows` instead. Otherwise, the test may silently pass if no exception is thrown.